newrelic/data_source_newrelic_account.go

```diff
@@ -60,7 +60,11 @@ func dataSourceNewRelicAccountRead(ctx context.Context, d *schema.ResourceData,
 	var account *accounts.AccountOutline
 
 	if !idOk && !nameOk {
-		return diag.FromErr(fmt.Errorf(`one of "name" or "account_id" is required to locate a New Relic account`))
+		id = meta.(*ProviderConfig).AccountID
```

Comment:
You might be able to use our helper function [`selectAccountID()`](https://github.com/newrelic/terraform-provider-newrelic/blob/main/newrelic/resource_helpers.go#L108-L124) for this.